lua/lspconfig/ui/lspinfo.lua

```diff
@@ -64,9 +64,13 @@ local function make_config_info(config, bufnr)
   local buffer_dir = api.nvim_buf_call(bufnr, function()
     return vim.fn.expand '%:p:h'
   end)
-  local root_dir = config.get_root_dir and config.get_root_dir(buffer_dir)
-  if root_dir then
-    config_info.root_dir = root_dir
+
+  if config.get_root_dir then
+    local root_dir
+    local status = coroutine.resume(coroutine.create(vim.schedule_wrap(function()
```

Comment:
I still don't get why `vim.schedule_wrap` is here, I think it's unnecessary and it only forces `get_root_dir` to be deferred. If I'm not mistaken this makes it so `root_dir` is always going to be `nil`, because it's assigned on the next event loop iteration once all of this code is done

Reply:
> I think it's unnecessary and it only forces get_root_dir to be deferred

that's right ..looks like before i don't get it work then i add it into schedule wrap . let me try again